src/platform/sdl/main.c

```diff
@@ -264,6 +278,12 @@ int mSDLRun(struct mSDLRenderer* renderer, struct mArguments* args) {
 					state->close(state);
 				}
 			}
+#ifdef USE_PLEDGE
+		if (!mPledgeNarrow(args)) {
```

Comment:
What's the purpose of this second, narrower call? To drop further privs after everything is initialized? It looks like all you drop is `dns` and `unix`, and I'm not sure what those are used for in the first place. SDL I guess. Note that the runloop will still get called if this fails.

Also the indentation doesn't match.

Reply:
Yes, that’s the purpose—pledge is intended to ratchet down permissions before entering the main loop.

`dns` and `unix` are needed for [audio](https://man.openbsd.org/sio_open.3#Use_with_pledge_2):

> * `unix` is needed when the device is a local sndiod(8) sub-device.
> * `inet` and `dns` are needed when the device is a remote sndiod(8) sub-device.
>
> Once no further calls to sio_open() will be made, all these pledge(2) promises may be dropped, except for the `audio` promise.

Reply:
Is the right thing here to return didFail right away and let the caller deinit everything?

Reply:
Everything after the runloop except for `mCoreThreadHasCrashed` still needs to be called.